includes/compat/trait-wc-stripe-subscriptions.php

```diff
@@ -1180,9 +1183,12 @@ protected function maybe_process_subscription_early_renewal_failure( $order, $in
 		if ( WC_Stripe_Subscriptions_Helper::is_subscriptions_enabled() && isset( $_GET['early_renewal'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification
 			$order->delete( true );
 			wc_add_notice( __( 'Payment authorization for the renewal order was unsuccessful, please try again.', 'woocommerce-gateway-stripe' ), 'error' );
-			$renewal_url = ( function_exists( 'wcs_get_early_renewal_url' ) && function_exists( 'wcs_get_subscription' ) )
-				? wcs_get_early_renewal_url( wcs_get_subscription( $order->get_meta( '_subscription_renewal' ) ) )
+
+			$subscription = WC_Stripe_Subscription::get_by_id( $order->get_meta( '_subscription_renewal' ) );
+			$renewal_url  = function_exists( 'wcs_get_early_renewal_url' ) && $subscription
+				? wcs_get_early_renewal_url( WC_Stripe_Subscription::get_by_id( $order->get_meta( '_subscription_renewal' ) ) )
```

Comment:
I think we need to pass the actual `WC_Subscription` object to this API.
```suggestion
			$renewal_url  = function_exists( 'wcs_get_early_renewal_url' ) && $subscription
				? wcs_get_early_renewal_url( $subscription->get_wc_subscription() ) // Not sure that we have this helper, but we probably need it!
```

Reply:
I forgot to replace the `wcs_get_early_renewal_url` parameter. I fixed it in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4254/commits/c3b314f177fa27ca6cee4c916dec561645e7d728 by passing the ID instead. The function accepts the ID, and it is preferable:
```php
function wcs_get_early_renewal_url( $subscription ) {
	$subscription_id = is_a( $subscription, 'WC_Subscription' ) ? $subscription->get_id() : absint( $subscription );

	$url = add_query_arg( array(
		'subscription_renewal_early' => $subscription_id,
		'subscription_renewal'       => 'true',
	), get_permalink( wc_get_page_id( 'myaccount' ) ) );

	/**
	 * Allow third-parties to filter the early renewal URL.
	 *
	 * @since 2.3.0
	 * @param string $url The early renewal URL.
	 * @param int    $subscription_id The ID of the subscription to renew to.
	 */
	return apply_filters( 'woocommerce_subscriptions_get_early_renewal_url', $url, $subscription_id ); // nosemgrep: audit.php.wp.security.xss.query-arg -- False positive. $url is escaped in the template and escaping URLs should be done at the point of output or usage.
}
```